### ISSUE TITLE
Integrate MLflow to yolov5

### ISSUE THREAD

**Issue by kevin801221** [ultralytics/yolov5#11344]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

How to integrate my training process hyper-parameters on MLflow API?
Can I add some code to train.py? or something like that to do that??
I know currently is integration to WandB , but I want to know how to integrate on MLflow.
Thanks a lot.

**Comment by glenn-jocher** [comment]
@kevin801221, you can integrate your training hyper-parameters with MLflow by modifying the logging functions in train.py. First, import the mlflow library: `import mlflow`, and then initialize the run before starting the training loop: `mlflow.start_run()`.

When you log your metrics, you can log them to MLflow with `mlflow.log_metric(name, value)`. You can also log hyperparameters with `mlflow.log_param(name, value)`. Don't forget to end the run after the training loop finishes with `mlflow.end_run()`.

Here's an example of how to log hyperparameters like learning rate and batch size to MLflow:

```python
# Import mlflow library
import mlflow

# Start a run
mlflow.start_run()

# Log hyperparameters
mlflow.log_param("learning_rate", 0.001)
mlflow.log_param("batch_size", 16)

# Training loop
for epoch in range(epochs):
   # ...

# End the run
mlflow.end_run()
```

For more details, please refer to the MLflow documentation: https://mlflow.org/docs/latest/tracking.html.

**Comment by kevin801221** [comment]
Hi glenn-jocher, thanks for your response.

**Comment by glenn-jocher** [comment]
You're welcome, @kevin801221! If you have any further questions or issues, please don't hesitate to ask. We're here to help!

**Comment by minazamani7** [comment]
Hi glenn-jocher, where should we add mlflow.sklearn.log_model() ?
